### PR TITLE
add Kubeconfig

### PR DESCRIPTION
**Pull request by fmoral2**
#### Proposed Changes ####

- https://confluence.suse.com/display/RKE2K3S/Add+existent+KubeConfig+on+test+framework
- https://github.com/rancher/qa-tasks/issues/862 

#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

https://github.com/rancher/distros-test-framework/pull/128#issuecomment-2231887499 

2. Verify code lint; we should not have errors.

ok 

3. Update the documentation if needed.

ok 

4. Update makefile and docker run if adding new tests.

n/a

5. Run your tests at least 4 times with all configurations needed and possible.

ok 
6. If needed test with different os types.

ok

### CODE REVIEW

**Comment by fmoral2** [comment]
jenkins PR  https://github.com/rancherlabs/jenkins-job-builder/pull/477

File: docs/testing.md [hunk 1]
```diff
@@ -114,3 +114,12 @@ go test -timeout=30m -v -tags=deployrancher ./entrypoint/deployrancher/... \
...
 chartsArgs rancherImage=<image or url>,extraEnv[0].name=CATTLE_AGENT_IMAGE,extraEnv[0].value=<image or url>-agent:v2.7.12
 ```
+
+### Testing with kubeconfig file.
+
+- Required variables in `.env` file
+```
+KUBE_CONFIG=<kubeconfig file 64 base encoded>
+BASTION_IP=<bastion public ip> when testing Dual-Stack
+AWS_USER_LOGIN=<aws user login> for ssh operations needed by `RunCommandOnNode()`
```
Comment (mdrahman-suse) [inline docs/testing.md:124]: Can we re-purpose `aws_user` variable from tfvars file?

Comment (fmoral2) [reply docs/testing.md:124]: people can use their preference OS, meaning i dont need to run in the same Os, so we need to ask that info

File: pkg/testcase/certrotate.go [hunk 2]
```diff
@@ -4,17 +4,16 @@ import (
...
 	. "github.com/onsi/gomega"
 )
 
-func TestCertRotate(cluster *factory.Cluster) {
+func TestCertRotate(cluster *shared.Cluster) {
 	certRotate(cluster.Config.Product, cluster.ServerIPs)
 
 	ip, manageError := shared.ManageService(cluster.Config.Product, "restart", "agent", cluster.AgentIPs)
-	Expect(manageError).NotTo(HaveOccurred(), fmt.Sprintf("error restarting agent node ip %s", ip))
+	Expect(manageError).NotTo(HaveOccurred(), "error restarting agent node ip"+ip)
```
Comment (mdrahman-suse) [inline pkg/testcase/certrotate.go:16]: Should add a space here

Comment (fmoral2) [reply pkg/testcase/certrotate.go:16]: no need here, the compiler automatically does that 
Screenshot 2024-08-05 at 16 08 44

**Review by mdrahman-suse — COMMENTED** [review]
Few comments, else LGTM

File: docs/testing.md (hunk 1 above)
Comment (mdrahman-suse) [reply docs/testing.md:124]: I mean we are passing the `aws_user` in the tfvars when running this, right? Can't we re-user that var in the test?

File: config/reader.go [hunk 3]
```diff
@@ -60,10 +61,17 @@ func loadEnv() (*Product, error) {
 		os.Exit(1)
 	}
 
+	// set the environment variables from the tfvars file.
```
Comment (fmoral2) [inline config/reader.go:64]: .

File: docs/testing.md (hunk 1 above)
Comment (fmoral2) [reply docs/testing.md]: totally true
fixed

File: docs/testing.md [hunk 4]
```diff
@@ -114,3 +114,11 @@ go test -timeout=30m -v -tags=deployrancher ./entrypoint/deployrancher/... \
 ```
 chartsArgs rancherImage=<image or url>,extraEnv[0].name=CATTLE_AGENT_IMAGE,extraEnv[0].value=<image or url>-agent:v2.7.12
 ```
+
+### Testing with kubeconfig file.
+
+- Required variables in `.env` file
+```
+KUBE_CONFIG=<kubeconfig file 64 base encoded>
```
Comment (mdrahman-suse) [inline docs/testing.md:122]: `base64-encoded`

**Review by mdrahman-suse — DISMISSED** [review]
One minor typo, else LGTM

**Review by mdrahman-suse — APPROVED** [review]
LGTM 🎉

**Review by rancher-max — APPROVED** [review]

**Pull request merged**